app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -100,6 +100,14 @@ def storage_account_name
     source.description.split("\\")[1]
   end
 
+  def invalid_or_nil_floating_ip?
+    floating_ip.nil? || floating_ip.network_port.nil?
```

Comment:
maybe use `try!` here and make this a positive method?

```ruby
def floating_ip_with_network_port_present?
  floating_ip.try!(:network_port).present?
end
```

or maybe just inline that above:
```ruby
nic_id = floating_ip.try!(:network_port).present? associated_nic : create_nic
```